server/test/controllers/applicant/ApplicantProgramBlocksControllerTest.java

```diff
@@ -751,6 +792,134 @@ public void update_requestedActionReview_answersSaved() {
+                    program.id,
+                    /* blockId= */ "4",
+                    /* inReview= */ false,
+                    new ApplicantRequestedActionWrapper(ApplicantRequestedAction.PREVIOUS_BLOCK)))
+            .bodyForm(
+                ImmutableMap.of(
+                    Path.create("applicant.applicant_email_address.email").toString(),
+                    "test@gmail.com"))
+            .build();
+
+    Result result =
+        subject
+            .updateWithApplicantId(
+                request,
+                applicant.id,
+                program.id,
+                /* blockId= */ "4",
+                /* inReview= */ false,
+                new ApplicantRequestedActionWrapper(ApplicantRequestedAction.PREVIOUS_BLOCK))
+            .toCompletableFuture()
+            .join();
+
+    assertThat(result.status()).isEqualTo(SEE_OTHER);
+    String previousRoute =
+        routes.ApplicantProgramBlocksController.previous(
+                // The 4th block was filled in, which is index 3. So, the previous block would be
+                // index 2.
+                program.id, /* previousBlockIndex= */ 2, /* inReview= */ false)
+            .url();
+    assertThat(result.redirectLocation()).hasValue(previousRoute);
+  }
+
+  @Test
+  public void update_requestedActionPrevious_isFirstBlock_redirectsToReview() {
+    program =
+        ProgramBuilder.newActiveProgram()
+            .withBlock("block 1")
+            .withRequiredQuestion(testQuestionBank().applicantName())
+            .build();
+    Request request =
+        requestBuilderWithSettings(
+                routes.ApplicantProgramBlocksController.updateWithApplicantId(
+                    applicant.id,
+                    program.id,
+                    /* blockId= */ "1",
+                    /* inReview= */ false,
+                    new ApplicantRequestedActionWrapper(ApplicantRequestedAction.PREVIOUS_BLOCK)))
+            .bodyForm(
+                ImmutableMap.of(
+                    Path.create("applicant.applicant_name").join(Scalar.FIRST_NAME).toString(),
+                    "FirstName",
+                    Path.create("applicant.applicant_name").join(Scalar.LAST_NAME).toString(),
+                    "LastName"))
+            .build();
+
+    Result result =
+        subject
+            .updateWithApplicantId(
+                request,
+                applicant.id,
+                program.id,
+                /* blockId= */ "1",
+                /* inReview= */ false,
+                new ApplicantRequestedActionWrapper(ApplicantRequestedAction.PREVIOUS_BLOCK))
+            .toCompletableFuture()
+            .join();
+
+    assertThat(result.status()).isEqualTo(SEE_OTHER);
+    String reviewRoute = routes.ApplicantProgramReviewController.review(program.id).url();
+    assertThat(result.redirectLocation()).hasValue(reviewRoute);
+  }
+
+  @Test
+  public void update_requestedActionPrevious_answersSaved() {
```

Comment:
**question**: If the first block redirects to the review page, would we want to test that answers are saved on block 2 or 3 instead? Or is that redirect after any saving logic?

Reply:
That redirect is after any saving logic but I think it's a good idea to add a test for it in case we change the flow on the backend. Added.